source/common/http/conn_manager_impl.cc

```diff
@@ -447,6 +455,12 @@ void ConnectionManagerImpl::ActiveStream::onIdleTimeout() {
   }
 }
 
+void ConnectionManagerImpl::ActiveStream::onRequestTimeout() {
+  connection_manager_.stats_.named_.downstream_rq_path_timeout_.inc();
+  sendLocalReply(request_headers_ != nullptr && Grpc::Common::hasGrpcContentType(*request_headers_),
```

Comment:
Can we ASSERT here that we have not started any response?

Reply:
Did you mean to add that boolean check within my call to sendLocalReply, or to add the assertion to the function itself?

Also, does this check conflict with disarming on encodeHeaders?

Reply:
doneeee